### PR TITLE
Add all uint/int/bytes primitive types for EIP-712 signing

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

This adds all uint/int/bytes primitive types for EIP-712 signing.

Swapping on Uniswap with Valora was failing with the following error: `Unrecognized type uint160 is not included in the EIP-712 type list`

Uniswap started using [Permit2](https://docs.uniswap.org/contracts/permit2/overview) which requires signing a message containing `uin160` and `uint48` types which were missing in the implementation.

See also https://github.com/ethereum/go-ethereum/pull/26770

### Other changes

Also added support for all the added types for `zeroValue`

### Tested

- Update unit tests
- Swapping on Uniswap with Valora works with the change

### Related issues

N/A

### Backwards compatibility

Yes

### Documentation

N/A

### CODE REVIEW

File: packages/sdk/utils/src/sign-typed-data-utils.test.ts [hunk 1]
```diff
@@ -268,6 +268,169 @@ const TEST_TYPES: EIP712TestCase[] = [
       },
     ],
   },
+  {
+    primaryType: 'AllAtomicTypes',
+    types: {
+      AllAtomicTypes: [
```
Comment (jeanregisser) [inline packages/sdk/utils/src/sign-typed-data-utils.test.ts:274]: I know I could have used a loop to generate those, but figured being more explicit was better in this testing context.

File: packages/sdk/utils/src/sign-typed-data-utils.ts [hunk 2]
```diff
@@ -40,14 +40,17 @@ export interface EIP712TypedData {
...
+  // and uint/int types from 8 to 256 bits
+  ...(() => {
+    const result = []
+    for (let i = 1; i <= 32; i++) {
+      result.push('bytes' + i)
+      result.push('uint' + i * 8)
+      result.push('int' + i * 8)
+    }
+    return result
+  })(),
```
Comment (jeanregisser) [inline packages/sdk/utils/src/sign-typed-data-utils.ts]: This array is now bigger, but hopefully it shouldn't be a performance problem for finding values.

File: packages/sdk/utils/src/sign-typed-data-utils.ts [hunk 3]
```diff
@@ -59,9 +62,12 @@ export const EIP712_BUILTIN_TYPES = EIP712_ATOMIC_TYPES.concat(EIP712_DYNAMIC_TY
 // Regular expression used to identify and parse EIP-712 array type strings.
 const EIP712_ARRAY_REGEXP = /^(?<memberType>[\w<>\[\]_\-]+)(\[(?<fixedLength>\d+)?\])$/
 
-// Regular experssion used to identity EIP-712 integer types (e.g. int256, uint256, uint8).
+// Regular expression used to identity EIP-712 integer types (e.g. int256, uint256, uint8).
```
Comment (eelanagaraj) [inline packages/sdk/utils/src/sign-typed-data-utils.ts]: optional: if you don't mind while you're at it, `identity` -> `identify`

File: packages/sdk/utils/src/sign-typed-data-utils.ts (hunk 2 above)
Comment (eelanagaraj) [reply packages/sdk/utils/src/sign-typed-data-utils.ts]: Nice! Yeah agreed, can always optimize later (i.e. by ordering most common types first for faster short-circuiting)

**Review by eelanagaraj — APPROVED** [review]
nice!

File: packages/sdk/utils/src/sign-typed-data-utils.ts (hunk 3 above)
Comment (jeanregisser) [reply packages/sdk/utils/src/sign-typed-data-utils.ts]: Oh yeah, missed that 👍

File: packages/sdk/utils/src/sign-typed-data-utils.ts (hunk 2 above)
Comment (jeanregisser) [reply packages/sdk/utils/src/sign-typed-data-utils.ts]: Good idea about putting more common types first 👍 
I did a naive pass at this in https://github.com/celo-org/celo-monorepo/pull/10224/commits/4bcde56c434b7ce57546405a9c005e8d1372e3c2

**Review by aaronmgdr — APPROVED** [review]

**Pull request merged**